solutions/arrays/rearrange-array.js

```diff
@@ -0,0 +1,42 @@
+let fs = require("fs");
+let data = fs.readFileSync(0, 'utf-8');
+let idx = 0;
+data = data.split('\n');
+
+function readLine() {
+  idx++;
+  return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+// Use readLine() for taking input, it will read one line of from the input  and is stored in string format
+let arraySize = parseInt(readLine());
+
+let nums = [];
+for (let idx = 0; idx < arraySize; idx++) {
+    nums.push(parseInt(readLine()));
+}
+
+let indices = [];
+for (let idx = 0; idx < arraySize; idx++) {
+    indices.push(parseInt(readLine()));
+}
+
+let target = [];
+for (let idx = 0; idx < arraySize; idx++) {
+    let currElement = nums[idx];
+    let targetIndex = indices[idx];
+    // We need to insert currElement at targetIndex
+    // But, we cannot create additional space in the middle of an array
+    // So, we will first insert currElement at the end of target
+    // Then, we will shift it towards left, one index at a time
```

Comment:
@DevratAgarwal I changed the algorithm. I feel the new one is simpler and easier to understand. Also, this uses a constant auxiliary space (primary reason for change is that this is easier to understand).